iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/sofi.hpp

```diff
@@ -151,13 +151,15 @@ class SoFi
     uint64_t size() const noexcept;
 
   private:
+    // safe access is guaranteed since the array is wrapped inside the SoFi
+    // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
     ValueType m_data[INTERNAL_SOFI_SIZE];
     uint64_t m_size = INTERNAL_SOFI_SIZE;
 
     /// @brief the write/read pointers are "atomic pointers" so that they are not
     /// reordered (read or written too late)
-    std::atomic<uint64_t> m_readPosition{0u};
-    std::atomic<uint64_t> m_writePosition{0u};
+    std::atomic<uint64_t> m_readPosition{0};
```

Comment:
Why removing the `U`? We should always try to be as explicit as possible.

Reply:
Why is this statement more explicit with a `U` suffix? Anyway it's not necessary here, please see https://github.com/eclipse-iceoryx/iceoryx/discussions/617.

Reply:
In my opinion the `U` suffix does not make the code safer, faster or improve the readability and even a google search did not reveal a coding guideline which universally enforces this. From the first look not even misra or autosar.

So I would remove `U` and make our life simpler.

Reply:
@FerdinandSpitzschnueffler @mossmaurice 

Yes remove it, especially for `0` and `1`. I explained some of the conversion rules here https://github.com/eclipse-iceoryx/iceoryx/discussions/617#discussioncomment-3093430. Feel free to correct any wrong information.

It is important to note that for any `intX_t` and the corresponding `uintX_t` the bit representation of all values up to `2^(X-1)-1` is identical. That includes literals.

It is also the case that for X < Y the bit representation of any `intX_t` and `intY_t` for all values up to `2^(X-1)-1` (the max for `intX_t` is the same if we ignore leading zeros. Therefore it is no problem to assign any value (literal or otherwise) of a smaller int type to a larger one - integer promotion is well defined.

This is the vast majority of cases (if not all) where we currently use `U`, it makes no difference to the compiler and produces the same code. All it does is add noise. 

Also note that technically for `uint64_t` the corresponding literal type is `ULL`. But again, this makes no difference as long as we are not even close to needing the extra bits to represent the number (it is promoted/converted and according to the rules).